continue
p3
p3.errors
